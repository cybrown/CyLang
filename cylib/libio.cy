void extern putchar(int c);
void extern putbyte(byte c);
void extern putshort(short c);
void extern putint(int c);
void extern putlong(long c);
void extern putfloat(float f);
void extern putdouble(double d);
void extern puts(byte[] str);
void extern printf(byte[] str);

void print(byte c) {
	putbyte(c);
}

void print(short c) {
	putshort(c);
}

void print(int c) {
	putint(c);
}

void print(long c) {
	putlong(c);
}

void print(float f) {
	putfloat(f);
}

void print(double d) {
	putdouble(d);
}

void print(byte[] str) {
	printf(str);
}

void print(bool c) {
	print(c ? "true" : "false");
}
