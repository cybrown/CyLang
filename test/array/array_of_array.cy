import libio;

int[2] arr0;
int[2] arr1;

int[2][2] arr;
int[2][2] arre;
int[2][2] arree;

arr[0] = arr0;
arr[1] = arr1;

// Testing successive affectation of arrays
arree = arre = arr;

arr[0][1] = 42;
arr[1][0] = 64;
arr[1][1] = 1515;

putint(arree[0][1]); putchar(10);
putint(arree[1][0]); putchar(10);
putint(arree[1][1]); putchar(10);
putchar(10);

putint(arr0[1]); putchar(10);
putint(arr1[0]); putchar(10);
putint(arr1[1]); putchar(10);
putchar(10);
