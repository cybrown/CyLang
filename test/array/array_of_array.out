42
64
1515

42
64
1515

